parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -292,11 +317,39 @@ public <D> FileAppender<D> build() throws IOException {
             .withRowGroupSize(rowGroupSize)
             .withPageSize(pageSize)
             .withDictionaryPageSize(dictionaryPageSize)
-            .build(),
+            // TODO: add .withMaxBloomFilterBytes(bloomFilterMaxBytes) once ParquetWriter.Builder supports it
+            .withBloomFilterEnabled(bloomFilterEnabled);
+
+        for (Map.Entry<String, String> entry : columnBloomFilterEnabled.entrySet()) {
+          String colPath = entry.getKey();
+          String bloomEnabled = entry.getValue();
+          parquetWriteBuilder.withBloomFilterEnabled(colPath, Boolean.valueOf(bloomEnabled));
+        }
+
+        for (Map.Entry<String, String> entry : columnBloomFilterNDVs.entrySet()) {
+          String colPath = entry.getKey();
+          String numDistinctValue = entry.getValue();
+          parquetWriteBuilder.withBloomFilterNDV(colPath, Long.valueOf(numDistinctValue));
+        }
+
+        return new ParquetWriteAdapter<>(
```

Comment:
@aokolnychyi, what do you think about removing the old `ParquetWriteAdapter` code? I don't think that anyone uses it anymore.